syntax = "proto3";
option java_multiple_files = true;
option java_package = "scanit.ocr.receipt.proto";
option java_outer_classname = "ScanitOCRServiceProto";
package scanit.ocr.receipt.proto;

message Error {
	int32 error_code  		= 1; /* Mã lỗi */
	string error_message 	= 2; /* Thông tin lỗi */
}

message AppCheckRequest {
	repeated string images		= 1; /* Ảnh định dạng base64 */	
	string request_id	= 2; /* Request id */
}

message AppCheckResponse {
	Error error			= 1; /* Thông tin lỗi */
	string json_data	= 2; /* Dữ liệu appcheck được */
    string metadata     = 3; /* Metadata */
}

message OCRRequest {	
	string request_id	= 1; /* Request id */
}

message OCRResponse {
	Error error			= 1; /* Thông tin lỗi */
	string json_data	= 2; /* Dữ liệu ocr được */
    string metadata     = 3; /* Metadata */
}

service ScanitOCRServices {
	rpc AppCheck (AppCheckRequest) returns (AppCheckResponse) {}
	rpc OCR (OCRRequest) returns (OCRResponse) {}
}

